### DISCUSSION TITLE
How to get all repo commits?

### DISCUSSION THREAD

**Discussion by ossama0808** [SpinlockLabs/github.dart#261] [Q&A] [answered]
Hi everyone,
I'm trying to get all commits in one single repo, is there a way to do this?

**Comment by robrbecker** [comment, marked as answer]
You would use the `listCommits` method and then paginate through results as needed. https://pub.dev/documentation/github/latest/github/RepositoriesService/listCommits.html

Although this will by default list ALL commits (probably in ascending order) which could be a LOT of data. Looking at the API, there's also no way to easily include parameters to get the commits you want (as seen here https://docs.github.com/en/rest/reference/repos#list-commits--parameters)

I'll make an issue to add the ability to pass in params to this method.

**Reply by robrbecker** [reply]
https://github.com/SpinlockLabs/github.dart/issues/263
